### DISCUSSION TITLE
It is not generating image

### DISCUSSION THREAD

**Discussion by hayashi-sensei** [nftchef/art-engine#277] [Help + Q&A] [answered]
PS D:\CodeProject\art-engine> node index.js
Debugger listening on ws://127.0.0.1:50112/34f7808a-3d28-40a9-bf6a-1dba06bd9664
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
D:\CodeProject\art-engine\src\config.js
genator started undefined
Editions left to create:  [
   1,  2,  3,   4,  5,  6,  7,  8,  9, 10, 11, 12,
  13, 14, 15,  16, 17, 18, 19, 20, 21, 22, 23, 24,
  25, 26, 27,  28, 29, 30, 31, 32, 33, 34, 35, 36,
  37, 38, 39,  40, 41, 42, 43, 44, 45, 46, 47, 48,
  49, 50, 51,  52, 53, 54, 55, 56, 57, 58, 59, 60,
  61, 62, 63,  64, 65, 66, 67, 68, 69, 70, 71, 72,
  73, 74, 75,  76, 77, 78, 79, 80, 81, 82, 83, 84,
  85, 86, 87,  88, 89, 90, 91, 92, 93, 94, 95, 96,
  97, 98, 99, 100,
  ... 100 more items
]
DNA: [
  '0.4:5.png',
  '0.3:4.png',
  '0.2:3.png',
  '0.1:2.png',
  '0.0:1.png',
  '1.9:flipping_w.png',
  '1.8:flipping_n.png',
  '1.7:flipping_l.png',
  '1.6:flipping_h.png',
  '1.5:flipping_a.png',
  '1.4:fanning_w.png',
  '1.3:fanning_n.png',
  '1.2:fanning_l.png',
  '1.1:fanning_h.png',
  '1.0:fanning_a.png',
  '1.34:meditate_w.png',
  '1.33:meditate_n.png',
  '1.32:meditate_l.png',
  '1.31:meditate_h.png',
  '1.30:meditate_a.png',
  '1.29:holding_w.png',
  '1.28:holding_n.png',
  '1.27:holding_l.png',
  '1.26:holding_h.png',
  '1.25:holding_a.png',
  '1.24:gloves_w.png',
  '1.23:gloves_n.png',
  '1.22:gloves_l.png',
  '1.21:gloves_h.png',
  '1.20:gloves_a.png',
  '1.19:gcup_w.png',
  '1.18:gcup_n.png',
  '1.17:gcup_l.png',
  '1.16:gcup_h.png',
  '1.15:gcup_a.png',
  '1.14:gbottle_w.png',
  '1.13:gbottle_n.png',
  '1.12:gbottle_l.png',
  '1.11:gbottle_h.png',
  '1.10:gbottle_a.png',
  '2.9:bokken.png',
  '2.8:baseballbat.png',
  '2.7:banner_seabream.png',
  '2.6:banner_prawn.png',
  '2.5:banner_koi.png',
  '2.4:banner_grey.png',
  '2.3:banner_daruma.png',
  '2.2:banner_crab.png',
  '2.1:banner_cat.png',
  '2.0:banner_brown.png',
  '2.49:unbrella_pink.png',
  '2.48:unbrella_peach.png',
  '2.47:talisman.png',
  '2.46:staff.png',
  '2.45:spearsblade.png',
  '2.44:spears.png',
  '2.43:shinai.png',
  '2.42:pokercard.png',
  '2.41:lantern_seabream.png',
  '2.40:lantern_prawn.png',
  '2.39:lantern_koi.png',
  '2.38:lantern_ebisu.png',
  '2.37:lantern_daruma.png',
  '2.36:lantern_crab.png',
  '2.35:lantern_cat.png',
  '2.34:kite_purple.png',
  '2.33:kite_green.png',
  '2.32:kite_blue.png',
  '2.31:katana_seabream.png',
  '2.30:katana_prawn.png',
  '2.29:katana_peachsky.png',
  '2.28:katana_peachribbon.png',
  '2.27:katana_peachblue.png',
  '2.26:katana_koi.png',
  '2.25:katana_daruma.png',
  '2.24:katana_crab.png',
  '2.23:katana_cat.png',
  '2.22:katana_black.png',
  '2.21:golfclub.png',
  '2.20:flipfan_yellow.png',
  '2.19:flipfan_pink.png',
  '2.18:fishingrod.png',
  '2.17:fan_purple.png',
  '2.16:fan_green.png',
  '2.15:coin_silver.png',
  '2.14:coin_bronze.png',
  '2.13:coffee_grey.png',
  '2.12:coffee_green.png',
  '2.11:card.png',
  '2.10:bottle.png',
  '3.2.4:male_w.png',
  '3.2.3:male_n.png',
  '3.2.2:male_l.png',
  '3.2.1:male_h.png',
  '3.2.0:male_a.png',
  '3.1.4:female_w.png',
  '3.1.3:female_n.png',
  '3.1.2:female_l.png',
  '3.1.1:female_h.png',
  '3.1.0:female_a.png',
  ... 474 more items
DNA exists!
DNA exists!
DNA exists!
DNA exists!
DNA exists!
DNA exists!
DNA exists!
You need more layers or elements to grow your edition to 100 artworks!
Waiting for the debugger to disconnect...
PS D:\CodeProject\art-engine> 

It only generates one image.
What should I do?

**Comment by PhosphoriaArtist** [comment, marked as answer]
Every image is being generated the exact same (same DNA), so it is ignoring duplicates. 
It looks like you don't have any '#' weightings on any of your pngs. So every image will contain every item which will result in every item being the exact same.

You need to have a weighting on every item in your folders (or on the folders themselves if you want everything in the folder to be drawn).
eg: 'female_l#20.png', 'female_h#20.png', 'female_a#20.png'
assuming the above images are in the same folder, that weighting gives each of them an equal chance to be selected (the number at the end can be anything, it's not a percent, it's relative to everything at the same folder level).
Anything without a weighting will be added to the image.

I suggest a mass renaming tool like ReNamer to add a `#20` to the end of every file name and try again.
